lib/matplotlib/animation.py

```diff
@@ -149,19 +142,28 @@ def is_available(self, name):
         Returns
         -------
         available : bool
-        '''
-        self.ensure_not_dirty()
-        return name in self.avail
-
-    def __getitem__(self, name):
-        self.ensure_not_dirty()
-        if not self.avail:
-            raise RuntimeError("No MovieWriters available!")
+        """
         try:
-            return self.avail[name]
+            cls = self._registered[name]
         except KeyError:
-            raise RuntimeError(
-                'Requested MovieWriter ({}) not available'.format(name))
+            return False
+        return cls.isAvailable()
+
+    def __iter__(self):
+        """Iterate over names of available writer class."""
+        for name in self._registered:
+            if self.is_available(name):
+                yield name
+
+    def list(self):
```

Comment:
Is this really necessary? Isn't `reg.list()` just the same as `list(reg)`?

Reply:
Oh, I see. This preserves API.

Reply:
Well, I can deprecate it too if you don't mind the API change...